api/server/secrets.go

```diff
@@ -0,0 +1,138 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+// TODO: Add swagger yaml
+func (c *clusterApi) setDefaultSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	method := "setDefaultSecretKey"
+	var secReq secrets.DefaultSecretKeyRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&secReq); err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err := c.SecretManager.SecretSetDefaultSecretKey(
+		secReq.DefaultSecretKey,
+		secReq.Override,
+	)
+
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// TODO: Add swagger yaml
+func (c *clusterApi) getDefaultSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	method := "getDefaultSecretKey"
+
+	secretValue, err := c.SecretManager.SecretGetDefaultSecretKey()
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	secResp := &secrets.GetSecretResponse{
+		SecretValue: secretValue,
+	}
+	json.NewEncoder(w).Encode(secResp)
+}
+
+// TODO: Add swagger yaml
+func (c *clusterApi) secretsLogin(w http.ResponseWriter, r *http.Request) {
+	var secReq secrets.SecretLoginRequest
+	method := "secretsLogin"
+
+	if err := json.NewDecoder(r.Body).Decode(&secReq); err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err := c.SecretManager.SecretLogin(secReq.SecretType, secReq.SecretConfig)
+	if err != nil {
+		if err == secrets.ErrNotAuthenticated {
+			c.sendError(c.name, method, w, err.Error(), http.StatusUnauthorized)
+			return
+		}
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// TODO: Add swagger yaml
+func (c *clusterApi) setSecret(w http.ResponseWriter, r *http.Request) {
+
+	method := "setSecret"
+	var secReq secrets.SetSecretRequest
+	params := r.URL.Query()
+	secretID := params[secrets.SecretKey]
+
+	if len(secretID) == 0 || secretID[0] == "" {
+		c.sendError(c.name, method, w, "Missing secret ID", http.StatusInternalServerError)
```

Comment:
This should be a status error of 400 (invalid requets) not a server error of 500 (StatusInternalServerError is a HTTP error) since it was caused by the client. See https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

Reply:
updated